packages/context/src/__tests__/acceptance/class-level-bindings.acceptance.ts

```diff
@@ -458,8 +458,8 @@ describe('Context bindings - Injecting dependencies of classes', () => {
         'location',
         {},
         // Set up a custom resolve() to access information from the session
-        (c: Context, injection: Injection, session: ResolutionSession) => {
-          resolutionPath = session.getResolutionPath();
+        (c: Context, injection: Injection, session?: ResolutionSession) => {
+          resolutionPath = session ? session.getResolutionPath() : undefined;
```

Comment:
I think this is showing a problem in the definition of `ResolveFunction` type. Is it a valid scenario that the resolver is called with no session argument? I though that we always create a new resolution session before calling any resolvers. There are few more places in our tests where I have to add an explicit handling of no resolution session, see below.

WDYT?

Reply:
Let's make `session` required. Can you open a PR to address that first?

Reply:
Sure, see https://github.com/strongloop/loopback-next/pull/2711

Reply:
Inspired by this comment, I also created https://github.com/strongloop/loopback-next/pull/2717.